emmarin/cl/cl/src/cl/bundle.rs

```diff
@@ -0,0 +1,117 @@
+use serde::{Deserialize, Serialize};
+
+use crate::cl::{partial_tx::PartialTx, BalanceWitness, PartialTxWitness};
+
+/// The transaction bundle is a collection of partial transactions.
+/// The goal in bundling transactions is to produce a set of partial transactions
+/// that balance each other.
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct Bundle {
+    pub partials: Vec<PartialTx>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct BundleWitness {
+    pub partials: Vec<PartialTxWitness>,
+}
```

Comment:
I think this struct shouldn't exist. We can't depend on any one entity having all partial tx witnesses for a bundle.

Reply:
Actually I don't think I'm using this anywhere. I've renamed the bundle proof to balance proof in the latest commit and would be perfectly fine to drop this struct for something like you proposed. RIght now `BalancePublic is the set of ptx balances

Reply:
yeah, I think we need anything else in the public statement of the bundle, just the bundle id should be enough.

Also, BalanceProof can commit to the BundleId rather than the set of balance commitments

Reply:
do we need to tie the bundle to the set of partialtxs instead of just balances? this could prevent some unbundling with a tx that has the same balance, but it's a bit more computation. We can also make the case that if a bundle is just about balances you can in fact swap out a ptx for another ptx of the same balance.
I mean that computing the bundle id might not be necessary and might take a few cycles more

Reply:
yeah, ok lets go with this for now, we can finesse it when we write the spec and thing through it more carefully.